src/cbc_sdk/platform/devices.py

```diff
@@ -315,6 +318,124 @@ def nsx_remediation(self, tag, set_tag=True):
+                          asset group query, or "MANUAL" to return only groups that the members were manually added to.
+                          Default is "ALL".
+
+        Returns:
+            list[str]: A list of asset group IDs this device belongs to.
+        """
+        rc = Device.get_asset_groups_for_devices(self._cb, self, **kwargs)
+        return rc[str(self._model_unique_id)]
+
+    def get_asset_groups(self, **kwargs):
+        """
+        Finds the list of asset groups that this device is a member of.
+
+        Required Permissions:
+            group-management(READ)
+
+        Parameters:
+            **kwargs (dict): Keyword arguments as documented below.
+
+        Keyword Args:
+            filter (str): Can restrict the types of group membership returned by this method.  Values are "ALL"
+                          to return all groups, "DYNAMIC" to return only groups that each member belongs to via the
+                          asset group query, or "MANUAL" to return only groups that the members were manually added to.
+                          Default is "ALL".
+
+        Returns:
+            list[AssetGroup]: A list of asset groups this device belongs to.
+        """
+        rc = Device.get_asset_groups_for_devices(self._cb, self, **kwargs)
+        return [self._cb.select("AssetGroup", v) for v in rc[str(self._model_unique_id)]]
+
+    @staticmethod
+    def _normalize_member_list(member_list):
+        """
+        Internal method which normalizes the parameters from add_members() or remove_members().
+
+        The method accepts ``Device`` objects (gets their ID), integers (converts them to strings), and simple
+        containers (recursively folds their contents into the return list). Everything else gets converted to
+        an integer, then to a string, and any value errors result in the value being silently dropped.
+
+        (This method was originally on AssetGroups, but it's needed by get_asset_groups_for_devices as well, and
+        having it here avoids the problem of circular imports.)
+
+        Args:
+            member_list (list[Any]): List of members to be normalized.
+
+        Returns:
+            list[str]: The normalized member list.
+        """
+        return_list = []
+        for m in member_list:
+            if m is None:
+                continue
+            if isinstance(m, Device):
+                return_list.append(str(m.id))
+            elif isinstance(m, int):
+                return_list.append(str(m))
+            elif isinstance(m, list) or isinstance(m, tuple) or isinstance(m, set):
+                return_list.extend(Device._normalize_member_list(m))
+            else:
+                try:
+                    return_list.append(str(int(m)))
+                except ValueError:
+                    pass
+        return return_list
+
+    @classmethod
+    def get_asset_groups_for_devices(cls, cb, *args, **kwargs):
```

Comment:
This is an advanced and complex function where as our target audience for the SDK is more beginner to moderate users. Why complex the interface? What problem are we trying to solve?